openpype/hosts/fusion/plugins/publish/collect_render.py

```diff
@@ -0,0 +1,207 @@
+            instance = FusionRenderInstance(
+                family="render",
+                tool=tool,
+                workfileComp=comp,
+                families=instance_families,
+                version=version,
+                time="",
+                source=current_file,
+                label=inst.data["label"],
+                subset=subset_name,
+                asset=inst.data["asset"],
+                task=task_name,
+                attachTo=False,
+                setMembers='',
+                publish=True,
+                name=subset_name,
+                resolutionWidth=comp_frame_format_prefs.get("Width"),
+                resolutionHeight=comp_frame_format_prefs.get("Height"),
+                pixelAspect=aspect_x / aspect_y,
+                tileRendering=False,
+                tilesX=0,
+                tilesY=0,
+                review="review" in instance_families,
+                frameStart=inst.data["frameStart"],
+                frameEnd=inst.data["frameEnd"],
+                handleStart=inst.data["handleStart"],
+                handleEnd=inst.data["handleEnd"],
+                ignoreFrameHandleCheck=(
+                    inst.data["frame_range_source"] == "render_range"),
+                frameStep=1,
+                fps=comp_frame_format_prefs.get("Rate"),
+                app_version=comp.GetApp().Version,
+                publish_attributes=inst.data.get("publish_attributes", {})
+            )
+
+            render_target = inst.data["creator_attributes"]["render_target"]
+
+            # Add render target family
+            render_target_family = f"render.{render_target}"
+            if render_target_family not in instance.families:
+                instance.families.append(render_target_family)
+
+            # Add render target specific data
+            if render_target in {"local", "frames"}:
+                instance.projectEntity = project_entity
+
+            if render_target == "farm":
+                fam = "render.farm"
+                if fam not in instance.families:
+                    instance.families.append(fam)
+                instance.toBeRenderedOn = "deadline"
+                instance.farm = True  # to skip integrate
+                if "review" in instance.families:
+                    # to skip ExtractReview locally
+                    instance.families.remove("review")
+
+            # add new instance to the list and remove the original
+            # instance since it is not needed anymore
+            instances.append(instance)
+            instances_to_remove.append(inst)
+
+        for instance in instances_to_remove:
+            context.remove(instance)
+
+        return instances
+
+    def post_collecting_action(self):
+        for instance in self._context:
+            if "render.frames" in instance.data.get("families", []):
+                # adding representation data to the instance
+                self._update_for_frames(instance)
```

Comment:
Admittedly the naming here is still confusing as it leaves so much more reading to understand why whatever is `render.frames` requires and update for frames?

```python
if instance.data.get("render_target") == "existing_frames":
    self._add_representations_from_expected_files(instance)
```
That to me would be more readable, but I'm fine with the current code if it works - but it gets me everytime that I still don't know why/what's being done here without continuing to read the rest of the code.